libstuff/SMultiHostSocketPool.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+#include <libstuff/STCPManager.h>
+#include <libstuff/SSocketPool.h>
+
+class SMultiHostSocketPool {
+  public:
+    SMultiHostSocketPool();
+    ~SMultiHostSocketPool();
+
+    // Returns an existing or new socket.
+    unique_ptr<STCPManager::Socket> getSocket(const string& host);
+
+    // Makes an existing socket available to be used again.
+    void returnSocket(unique_ptr<STCPManager::Socket>&& s, const string& host);
+
+  private:
+    mutex _poolMutex;
+    map<string, unique_ptr<SSocketPool>> _pools;
```

Comment:
I think this should just be `map<string, SSocketPool> _pools;` which would be simpler. I think this should work without worrying about duplicating the pool or anything by using `emplace` to insert new socket pools into `_pools`.